buildkite/build_release.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -xueo pipefail
+trap 'echo build script failed at line $LINENO' ERR
+
+TODAY=$(date +'%Y-%m-%d')
+: ${BRANCH:=master}
+
+if [ ! -d installer ]; then
+    git clone https://github.com/dlang/installer.git installer
+fi
+pushd installer/create_dmd_release
+
+git reset --hard --quiet
+git clean -df --quiet
+if [ $BRANCH != master ] && [ $BRANCH != stable ] &&
+       ! git ls-remote --exit-code --heads origin $BRANCH > /dev/null; then
+    # use master as fallback base for feature branches
+    git fetch origin master --quiet
+else
+    git fetch origin $BRANCH --quiet
+fi
+git checkout FETCH_HEAD --quiet
+
+mkdir build
+LATEST=$(curl -fsSL http://downloads.dlang.org/releases/LATEST)
+rdmd build_all.d v$LATEST $BRANCH && success=yes || success=no
+rm -f build/*.zip # only keep lzma archives
+
+for file in build/*; do
+    gpg --detach-sign --no-use-agent $file
+done
+
+mv build dmd-$BRANCH-$TODAY
+aws --profile ddo s3 sync dmd-$BRANCH-$TODAY/ s3://downloads.dlang.org/nightlies/dmd-$BRANCH-$TODAY/ --acl public-read --cache-control max-age=604800
```

Comment:
Should we really upload this automated into the same bucket? I mean the CI is a rather easy target vector and quite exposed.
Ideas:
1) use a different bucket or hoster (there are many free ones)
2) restrict the AWS token to the nightly folders only

Reply:
> 1. use a different bucket or hoster (there are many free ones)

Mmh, it makes a lot of sense to put it on the same download page as the other releases, see http://downloads.dlang.org/. It wasn't done originally as I didn't have access to ddo.

> 2. restrict the AWS token to the nightly folders only

That would be good, I'm already only synching the nightlies folder. Will ask Brad for another token.
Let's do this in a second iteration to not stall this. This is not intended to run as CI, but only via scheduler. Also I restricted buildkite access for the release-builder pipeline to admins, though nothing prevents people from scheduling other builds on that queue once the agent is running.

Reply:
> That would be good, I'm already only synching the nightlies folder. Will ask Brad for another token.

If the access is limited to a folder only, it should be fine. I was just worried that an accidental leak here could have an impact on all our binaries.

> Could prolly add an agent hook to verify we're on the right pipeline.
> ...
> Let's do this in a second iteration to not stall this. 

Sure. Let's just open an issue, s.t. this isn't lost?

Reply:
> Also I restricted buildkite access for the release-builder pipeline to admins, 

Be aware that the AWS credentials still might show up in the logs (or old logs) and currently "Everyone" has read access to the logs:

![image](https://user-images.githubusercontent.com/4370550/47293414-5fcbaa00-d60a-11e8-90f2-b660f39d0c97.png)

Reply:
See https://github.com/dlang/downloads.dlang.org/pull/5 to improve on this.